### ISSUE TITLE
to_json bug when using path and lazy options together

### ISSUE THREAD

**Issue by davlee1972** [sissaschool/xmlschema#159]
I was reviewing how to get JSONL implemented in to_json and found a bug

The following works:

```
my_dict = my_schema.to_dict("all.xml", validation='skip', path="/Feed/AssetOverview", lazy=True)
my_json = xmlschema.to_json("all.xml", schema=my_schema, validation='skip', path="/Feed/AssetOverview")
my_json = xmlschema.to_json("all.xml", schema=my_schema, validation='skip', lazy=True)
```

But when using path and lazy together in to_json nothing gets decoded:

```
my_json = xmlschema.to_json("all.xml", schema=my_schema,  validation='skip', path="/Feed/AssetOverview", lazy=True)
>>> my_json
'null'
```

**Comment by brunato** [comment]
Hi,
the bug seems to be limited to absolute paths. It lacks on calculating the right *path_level* for matching nodes into *XMLResource.iterfind()*.

I have added a fix and some tests based on collection samples into the next release (v1.1.0, Python 3-only).
Thanks

**Comment by brunato** [comment]
Hi @davlee1972,
this issue should be resolved in v1.1.0.

I've also added a new lazy decoder for JSON, that should be effective in memory saving. It works with two-step lazy iterations. The first pass create a decoding for lower levels of whole data (1-2 levels), filled with an iterator (*XMLSchema.raw_decoder()*). The second pass send the decoded structure to a json.dump/s with argument `cls=JSONLazyEncoder`.

I've also added a *ParquetConverter* class as experiment in *converters.py*. Take a look on this when you can.

Thanks

**Comment by brunato** [comment]
Hi @davlee1972 ,

the new converter will be added in next minor release (v1.2) soon. I'm oriented to find another name for it because it seems not only useful/related to Parquet format. I will add *element_encode()* to it to complete the decode/encode process at its best.

On JSONL I'm not oriented to add it to the package, because I don't want to add strict dependencies or write a new decoder/encoder on my own. For the next releases I will focus instead to improve the decoding process, maybe adding other options.

Thank you

**Comment by davlee1972** [comment]
ParquetConverter is probably not a good name.. How about ColumnarJSONConverter?

Eventually if JSONL gets added we can use both:

https://arrow.apache.org/docs/python/json.html to read JSONL

and then save columnar arrow memory objects into parquet once nested write supports gets added..

https://www.mail-archive.com/dev@arrow.apache.org/msg16763.html

**Comment by brunato** [comment]
*ColumnarJSONConverter* sounds good, but if this converter is not strictly related with JSON, what do you think of *ColumnarConverter*, that is  shorter and more general name?

**Comment by davlee1972** [comment]
Columnar converter sounds fine. Can't think or find anything else that conceptually fits better.

**Comment by brunato** [comment]
Hi @davlee1972,

the new converter is available with release v1.2.0.
Best
